Spatial_Engine/Create/ShapeProfiles/FreeFormProfile.cs

```diff
@@ -136,13 +138,29 @@ public static FreeFormProfile FreeFormProfile(IEnumerable<ICurve> edges)
                     }
                     if (intersects)
                         Engine.Reflection.Compute.RecordWarning("The Profiles curves are intersecting eachother.");
+
+                    try
+                    {
+                        //Try centering the curves around the origin;
+                        List<List<ICurve>> curves = result.ToList().IJoin().Cast<ICurve>().ToList().DistributeOutlines();
+                        Point centroid = curves.Select(x => x[0]).Centroid(curves.SelectMany(x => x.Skip(1)));
+                        Vector tranlation = Point.Origin - centroid;
+                        result = result.Select(x => x.ITranslate(tranlation)).ToList();
+                        couldCentre = true;
+                    }
+                    catch (Exception e)
+                    {
+
+                    }
                 }
                 // Is SelfInteersecting
                 if (joinedCurves.Any(x => Geometry.Query.IsSelfIntersecting(x)))
                     Engine.Reflection.Compute.RecordWarning("One or more of the Profiles curves is intersecting itself.");
-
             }
 
+            if (!couldCentre)
```

Comment:
Couldn't that sit in the `catch` from line 151? `couldCentre` could possibly be avoided then.

Reply:
was in case it is a nurbs or ellipse. Defined the variable outside the two if statements. So if it contains nurbs or ellipses, or crashes while trying, this will be false and raise the message, if that makes sense?

Reply:
Ah ok, makes perfect sense after a second look.